store/tikv/2pc.go

```diff
@@ -1456,7 +1454,7 @@ func (c *twoPhaseCommitter) calculateMaxCommitTS(ctx context.Context) error {
 }
 
 func (c *twoPhaseCommitter) shouldWriteBinlog() bool {
-	return c.txn.us.GetOption(kv.BinlogInfo) != nil
+	return c.txn.us.GetOption(kv.BinlogInfo) != nil && c.binlog != nil
```

Comment:
Maybe we could remove the option `kv.BinlogInfo` in the future, when the driver set the `kv.BinlogInfo` option, we init `c.binlog`

Reply:
good idea